### PR TITLE
add label permission to join clusterset

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>

### CODE REVIEW

File: pkg/webhook/cluster/validating_webhook.go [hunk 1]
```diff
@@ -292,3 +291,74 @@ func (a *ManagedClusterValidatingAdmissionHook) allowUpdateClusterSet(userInfo a
...
+	if status := a.allowJoinClusterSet(userInfo, clusterSetName); status.Allowed {
+		return status
+	}
+	// check "managedclusters/label" permission with resource name "labelKey:labelValue"
+	if status := a.allowUpdateLabel(userInfo, clusterSetLabel+":"+clusterSetName); status.Allowed {
+		return status
+	}
+
+	// check "managedclusters/label" permission with resource name "labelKey:*"
+	if status := a.allowUpdateLabel(userInfo, clusterSetLabel+":*"); status.Allowed {
```
Comment (qiujian16) [inline pkg/webhook/cluster/validating_webhook.go]: why we need this?

File: pkg/webhook/cluster/validating_webhook.go [hunk 2]
```diff
@@ -292,3 +291,74 @@ func (a *ManagedClusterValidatingAdmissionHook) allowUpdateClusterSet(userInfo a
...
+}
+
+// allowUpdateClusterSet check where a request user has been authorized to add/remove the clustersetLabel to/from ManagedCluster
+func (a *ManagedClusterValidatingAdmissionHook) allowUpdateClusterSet(userInfo authenticationv1.UserInfo, clusterSetName string) *admissionv1beta1.AdmissionResponse {
+	// check "managedclustersets/join" permission
+	if status := a.allowJoinClusterSet(userInfo, clusterSetName); status.Allowed {
+		return status
+	}
+	// check "managedclusters/label" permission with resource name "labelKey:labelValue"
+	if status := a.allowUpdateLabel(userInfo, clusterSetLabel+":"+clusterSetName); status.Allowed {
```
Comment (qiujian16) [inline pkg/webhook/cluster/validating_webhook.go]: to my memory, when check join we also check both if user can join old  and new clusterset.

File: pkg/webhook/cluster/validating_webhook.go (hunk 1 above)
Comment (ldpliu) [reply pkg/webhook/cluster/validating_webhook.go]: this is used to support permission like:
` - apiGroups: ["cluster.open-cluster-management.io"]
   resources: ["managedclusters/label"]
   resourceNames: ["cluster.open-cluster-management.io/clusterset:*"]
   verbs: ["create"]`
I am not sure if we should support it?

File: pkg/webhook/cluster/validating_webhook.go (hunk 2 above)
Comment (ldpliu) [reply pkg/webhook/cluster/validating_webhook.go]: Yes, we did the check in https://github.com/open-cluster-management-io/registration/pull/203/files#diff-042b72f0f7d4c234eb9f079b0fdfa71967b9f9c97dc0a121084177114ac3d1f6L226

This func is used to check one set's permission

File: pkg/webhook/cluster/validating_webhook.go (hunk 1 above)
Comment (qiujian16) [reply pkg/webhook/cluster/validating_webhook.go]: can you elaborate what '*' means here

File: pkg/webhook/cluster/validating_webhook.go (hunk 2 above)
Comment (qiujian16) [reply pkg/webhook/cluster/validating_webhook.go]: To my understanding, you should have a check func that look at the changed labels on the cluster, filter out the labels that is not used to do rbac check, then using the remaining labels to do rbac check

**Pull request closed without merging**